src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -305,6 +333,81 @@ private String postTestWebResource() throws Exception {
+
+        MvcResult mvcResult = mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().isCreated()).andReturn();
+
+        return mvcResult.getResponse().getHeader("Location");
+    }
+
+    private String postTestFTPWebResource() throws Exception {
+        WebResource testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "ftp://ftp.sra.ebi.ac.uk/meta/xsd/sra_1_5/SRA.study.xsd");
+
+        MvcResult mvcResult = mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().isCreated()).andReturn();
+
+        return mvcResult.getResponse().getHeader("Location");
+    }
+
+    private String postTestComplexWebResource() throws Exception {
+        WebResource testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "http://MVSXX.COMPANY.COM:04445/CICSPLEXSM//JSMITH/VIEW/OURLOCTRAN?A_TRANID=P*&O_TRANID=NE");
+
+        MvcResult mvcResult = mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().isCreated()).andReturn();
+
+        return mvcResult.getResponse().getHeader("Location");
+    }
+
+    @Test
+    public void postWebResourceInvalidURL() throws Exception {
+        WebResource testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "");
+
+        mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().is4xxClientError());
+
+        testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "htttps://www.ebi.ac.uk");
+
+        mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().is4xxClientError());
+
+        testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "www.google.com");
+
+        mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().is4xxClientError());
+
+        testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "http://www.space address.org");
+
+        mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().is4xxClientError());
+
+        testWebResource = new WebResource(WebResource.Type.CENTER_WEB, "//fileserver/code/src/main/app.java");
+
+        mockMvc.perform(post("/webResources")
+                .content(testWebResourceJson.write(testWebResource).getJson()))
+                .andExpect(status().is4xxClientError());
+
+
```

Comment:
Personally I would not leave that many blank lines here. There is a coding style guide here for your reference: https://www.ebi.ac.uk/seqdb/confluence/display/VAR/Java+coding+style